db/error_handler.h

```diff
@@ -53,13 +50,14 @@ class ErrorHandler {
                                      Status::Code code,
                                      Status::SubCode subcode);
 
-   Status SetBGError(const Status& bg_err, BackgroundErrorReason reason);
+   const Status& SetBGError(const Status& bg_err, BackgroundErrorReason reason);
```

Comment:
I see that because of some complex, possibly erroneous logic in StartRecoverFromRetryableBGIOError we can't equivalently make SetBGError void and have interested callers call GetBGError(). Can think about that later.